homeassistant/components/gree/switch.py

```diff
@@ -0,0 +1,81 @@
+"""Support for interface with a Gree climate systems."""
+import logging
+from typing import Optional
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_SWITCH,
+    DOMAIN as SWITCH_DOMAIN,
+    SwitchEntity,
+)
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Gree HVAC device from a config entry."""
+    async_add_entities(
+        [GreeSwitchEntity(device) for device in hass.data[DOMAIN].pop(SWITCH_DOMAIN)]
+    )
+
+
+class GreeSwitchEntity(CoordinatorEntity, SwitchEntity):
+    """Representation of a Gree HVAC device."""
+
+    def __init__(self, device):
+        """Initialize the Gree device."""
+        super().__init__(device)
+        self._device = device
+        self._name = device.device_info.name + " Panel Light"
+        self._mac = device.device_info.mac
+        self._icon = "mdi:lightbulb"
```

Comment:
Maybe just inline as return "mdi:lightbulb" in icon()? A member variable does not seem needed.